examples/bigcommerce/lib/makeswift/context.tsx

```diff
@@ -0,0 +1,149 @@
+import { createContext, ReactNode, useCallback, useContext, useEffect, useState } from 'react'
+
+export type Image = {
+  url: string
+  alt: string
+}
+export type Product = {
+  url: string
+  id: number
+  name: string
+  description: string
+  price: string
+  defaultImage: Image
+  images: Image[]
+  categories: Category[]
+}
+export type Category = 'Shop All' | 'Plants' | 'Pots'
+
+export const ProductsContext = createContext<any>(null)
+
+export const useProducts = (
+  {
+    count,
+    category,
+  }: {
+    count?: number
+    category?: Category
+  } = {
+    count: 4,
+    category: 'Shop All',
+  },
+): {
+  products: Product[]
+} => {
+  const products = useContext(ProductsContext)
+
+  return {
+    products: products
+      .map((product: any) => ({
+        url: `/product/${product.node.entityId}`,
+        id: product.node.entityId,
+        name: product.node.name,
+        price: product.node.prices.price.value.toFixed(2),
+        description: product.node.description,
+        defaultImage: {
+          url: product.node.defaultImage.url,
+          alt: 'You still need to specify',
+        },
+        images: product.node.images.edges.map(({ node }: any) => ({
+          url: node.url,
+          alt: 'You still need to specify',
+        })),
+        categories: product.node.categories.edges.map(({ node }: any) => ({
+          name: node.name,
+          path: node.path,
+        })),
+      }))
+      .filter((product: any) => product.categories.some((c: any) => c.name === category))
+      .slice(0, count),
+  }
+}
+export const ProductContext = createContext<any>(null)
+
+export const useProduct = (): {
+  product: Product
+} => {
+  const product = useContext(ProductContext)
+
+  return {
```

Comment:
We seem to use the exact same product mapping in the `useProducts` hook above. What do you think about creating a function that maps a `BigCommerceProduct` to this internal representation?